src/app/code/community/Shopgate/Cloudapi/Helper/Api2/Quote.php

```diff
@@ -44,13 +45,45 @@ public function addItemErrors(Mage_Sales_Model_Quote $quote, Mage_Core_Model_Sto
         Mage::app()->setCurrentStore($store->getCode());
         /** @var Mage_Sales_Model_Quote_Item $item */
         foreach ($quote->getAllItems() as $item) {
-            if ($item->getData('has_error')) {
-                $item->setData(self::KEY_ITEM_ERRORS, $item->getMessage(false));
+            if ($item->getData(self::KEY_ERRORS)) {
+                $item->setData(self::KEY_ERROR_MESSAGES, $item->getMessage(false));
             }
         }
         Mage::app()->setCurrentStore($adminStore);
     }
 
+    /**
+     * Adds errors to the quote.
+     * Store state switch is necessary as errors do not
+     * print in the Admin store state, which is the
+     * default global store endpoints run in.
+     *
+     * Currently only the minimum order amount will be validated.
+     *
+     * @param Mage_Sales_Model_Quote $quote
+     * @param Mage_Core_Model_Store  $store
+     *
+     * @throws Mage_Core_Model_Store_Exception
+     */
+    public function addQuoteErrors(Mage_Sales_Model_Quote $quote, Mage_Core_Model_Store $store)
+    {
+        $adminStore = Mage::app()->getStore();
+        Mage::app()->setCurrentStore($store->getCode());
+        /** @var Mage_Sales_Model_Quote_Item $item */
+        if (!$quote->validateMinimumAmount()) {
+            $minimumAmount = Mage::app()->getLocale()->currency($store->getCurrentCurrencyCode())
+                ->toCurrency(Mage::getStoreConfig('sales/minimum_order/amount'));
```

Comment:
throws an exception is value is empty, the validateMinimum should prevent this from happening I am assuming. My mage is currently broken so I cannot check. In any case, at least the throws tag should be annotated